app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,124 @@
+class RecipesController < ApplicationController
+  before_action :authenticate_user!, except: %i[index show]
+  before_action :set_recipe, only: %i[show edit update destroy]
+  before_action :authorize_user!, only: %i[edit update destroy]
+
+  def index
+    @recipes = Recipe.includes(:user).order(created_at: :desc)
```

Comment:
- [ ] It would be nice to display only the recipes created by the logged-in user
Example:
```rb
@recipes = current_user.recipes.order(created_at: :desc)
```